sandbox/src/Game.cpp

```diff
@@ -1,8 +1,27 @@
 #include "Game.hpp"
 
+std::unique_ptr<Yuna::Core::State>	stateFactory(uint8_t tID)
```

Comment:
This is an example of a state factory
This is useful because the state manager cannot know in advance what states will be added by the end user. This way the end user (the game developer) will be able to define what state to push. This can be further extended with an enum instead of IDs for more readability.

Reply:
nice!